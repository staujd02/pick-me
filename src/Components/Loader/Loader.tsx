import React from 'react';
import LoadScreen from '../LoadScreen/LoadScreen';
import ConfigureButton from '../Configuration/ConfigureButton';

class Loader extends React.Component<LoaderProps, LoaderState> {

    constructor(props: any) {
        super(props);
        this.state = {
            doneLoading: false
        }
        setTimeout(this.finishedLoading, this.props.loadTime);
    }

    finishedLoading = () => {
        this.setState({
            doneLoading: true
        });
    }

    render(){
        const { doneLoading } = this.state;
        const { handleConfigureClick } = this.props;
        if(doneLoading)
            return (
                <div>
                    {this.props.children}
                </div>
            );
        else
            return (
                <div>
                    <LoadScreen />
                    <ConfigureButton buttonClickHandler={handleConfigureClick} />
                </div>
            );
    }
}

export default Loader;