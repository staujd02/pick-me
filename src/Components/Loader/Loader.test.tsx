import React from 'react';
import Loader from './Loader';
import { shallow, ShallowWrapper } from 'enzyme';
import { formatHTML } from '../../TestUtilities/htmlFormatter';

describe('The Loader', () => {
    
    const element = <label>Yelp!</label>;
    const loadTime = 100;
    let wrapper: ShallowWrapper<LoaderProps, LoaderState, Loader>;
    let handleClick: jest.Mock;

    beforeEach(() => {
        let handleClick = jest.fn();
        wrapper = shallow(
            <Loader handleConfigureClick={handleClick} loadTime={loadTime}>
                {element}
            </Loader>)
    });

    it('renders correctly', () => expect(formatHTML(wrapper.html())).toMatchSnapshot());

    describe("given the load time has elapsed", () => {

        beforeEach(done => {
           setTimeout(done, loadTime); 
        });

        it("renders the loaded elements", () => {
            expect(wrapper.contains(element)).toEqual(true);
        });
    });

    describe('Given the configure button is still visible', () => {
        describe('When the user click the config button', () => {
            beforeEach(() => {
                (wrapper.find('ConfigureButton').props() as  ConfigureButtonProps)
                .buttonClickHandler({} as ButtonClickEvent);
            });

            it('reveals the triggers the props callback', () => {
                expect(handleClick).toHaveBeenCalled();
            }); 
        });
    });
    
    describe("given the load time has not elapsed", () => {
        it("doesn't renders the loaded elements", () => {
            expect(wrapper.contains(element)).toEqual(false);
        });
    });

});