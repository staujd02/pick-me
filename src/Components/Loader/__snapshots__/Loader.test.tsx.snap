// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Loader renders correctly 1`] = `
"
<div>
  <header class=\\"App-header\\">
    <img src=\\"kaleidoscope-2.svg\\" class=\\"App-logo\\" alt=\\"logo\\">
    <p>Starting Up The Streams...</p>
  </header>
  <button>Configure Streams</button>
</div>
"
`;
